File: problem-solving/baekjoon/11866_요세푸스_문제_0.go
// https://www.acmicpc.net/problem/11866
// 큐를 이용해 제거 과정을 구현하는 문제
package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	writer := bufio.NewWriter(os.Stdout)
	defer writer.Flush()

	var n, k int
	fmt.Fscanln(reader, &n, &k)

	var queue, result []int
	for i := 1; i <= n; i++ {
		queue = append(queue, i)
	}

	var delIdx int
	for len(queue) > 0 {
		delIdx = (delIdx + k - 1) % len(queue)
		result = append(result, queue[delIdx])
		queue = append(queue[:delIdx], queue[delIdx+1:]...)
	}

	for i, v := range result {
		if i == 0 {
			fmt.Fprintf(writer, "<%d", v)
		} else {
			fmt.Fprintf(writer, ", %d", v)
		}
	}
	fmt.Fprintln(writer, ">")
}

File: problem-solving/baekjoon/11866_요세푸스_문제_0.py
# https://www.acmicpc.net/problem/11866
# 큐를 이용해 제거 과정을 구현하는 문제
import sys
from collections import deque

if __name__ == "__main__":
    n, k = list(map(int, sys.stdin.readline().split()))

    queue = deque(maxlen=n)
    for i in range(1, n+1):
        queue.append(i)
    
    result = []
    del_idx = 0
    while len(queue) > 0:
        del_idx = (del_idx+k-1)%len(queue)
        result.append(queue[del_idx])
        del queue[del_idx]
    
    for i, v in enumerate(result):
        if i == 0:
            print("<{}".format(v), end='')
        else:
            print(", {}".format(v), end='')
    print(">")